libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/factory/DropInUIMapboxNavigationFactory.kt

```diff
@@ -0,0 +1,24 @@
+package com.mapbox.navigation.dropin.internal.factory
+
+import android.content.Context
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.MapboxNavigationProvider
+
+class DropInUIMapboxNavigationFactory(
+    private val context: Context,
+    private val accessToken: String
+) {
+
+    fun getMapboxNavigation(): MapboxNavigation {
+        return if (MapboxNavigationProvider.isCreated()) {
```

Comment:
This will get resolved with the lifecycle work that @kmadsen has been working on. I think in the short term this PR should get merged in as is so we can move forward with other work. Once the lifecycle observer mentioned previously is merged I can refactor the necessary code to use it.

Reply:
Getting the ball rolling over here https://github.com/mapbox/mapbox-navigation-android/pull/5112

Reply:
I agree that Drop-In should have full control over the `MapboxNavigation` instance. We should use just  `MapboxNavigationProvider#create` here and not check for any existing instances before. That's the better setup for now while we continue the discussion about the general lifecycle separately.